### DISCUSSION TITLE
Remove workflows older than xxx

### DISCUSSION THREAD

**Discussion by pierre-gautier** [community/community#26044] [Actions]
<p>Hi,</p>
<p>my old github action workflows (more than 600 atm) are still referencing commits that are not referenced by any branch and should normally be unreachable and thus deleted. Maybe the workflow implementation is using underlying branches.</p>
<p>Your ways of managing data storage is none of my business but it actually causes me some pain since the jira integration shows all the commits containing reference to the ticket … even from unreachable commits that have triggered workflows, have been changed and are no longer relevant.</p>
<p>So, as many of us have done before me, I’m begging you to implement an API to remove old executed workflows so we can clean them up like developers do : not manually !</p>
<p>By advance, I thank you for this and for the action workflows that otherwise is perfect for my usage !</p>
<p>Pierre</p>

**Comment by chrispat** [comment, marked as answer]
<p>You can find the documentation for the API to delete a workflow run on the GitHub Docs <a href="https://docs.github.com/en/rest/reference/actions#delete-a-workflow-run" rel="nofollow noopener">https://docs.github.com/en/rest/reference/actions#delete-a-workflow-run</a>.</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/pierre-gautier">@pierre-gautier</a>,</p>
<p>You can complete this with two apis:</p>
<p>Get all the workflow runs for your repository with this <a href="https://docs.github.com/en/rest/reference/actions#list-workflow-runs-for-a-repository" rel="nofollow noopener">API</a>, add parameters to filter:</p>
<pre><code class="lang-auto">curl -H "Accept: application/vnd.github.v3+json" "https://api.github.com/repos/{org}/{repo}/actions/runs?per_page=100&amp;page=5"  # list page5, per-page 100
</code></pre>
<p>parse the workflow run id which timestamp is older than XXX(eg: use jq command).</p>
<p>Create a loop, use the workflow run ids for deletion workflow runs API:<br>
<a href="https://docs.github.com/en/rest/reference/actions#delete-a-workflow-run" class="onebox" target="_blank" rel="nofollow noopener">https://docs.github.com/en/rest/reference/actions#delete-a-workflow-run</a></p>
<p>Thanks</p>

**Comment by pierre-gautier** [comment]
<p>Thanks a lot !</p>
<p>I did some researches before posting but all I could find was this <a href="https://github.blog/changelog/2020-07-07-github-actions-ability-to-delete-workflow-runs/" rel="nofollow noopener">blog entry</a> and did not think to check for the API doc, my bad.</p>
<p>Since others could follow the same path, maybe adding something in this blog entry would be a good idea.</p>
<p>Last but not least, it seems to me that most of github actions users will need an action like <a href="https://github.com/marketplace/actions/purge-artifacts" rel="nofollow noopener">purge artifacts</a> to clean runs one day or another. Something even better integrated like configuration would be even greater.</p>
<p>Have a good day</p>
